processing/src/main/java/io/druid/query/extraction/MatchingDimExtractionFn.java

```diff
@@ -59,9 +60,8 @@ public MatchingDimExtractionFn(
   @Override
   public String apply(String dimValue)
   {
-    dimValue = (dimValue == null) ? "" : dimValue;
-    Matcher matcher = pattern.matcher(dimValue);
-    return matcher.find() ? dimValue : null;
+    Matcher matcher = pattern.matcher(Strings.nullToEmpty(dimValue));
```

Comment:
yeah, that's a potential optimization. The bug being fixed was that a pattern matching empty string would return `""` on `""` but it should actually return `null`. (the new test checks for that)